src/new_hope/week3.clj

```diff
@@ -0,0 +1,148 @@
+(ns new-hope.week3
+  (:require [midje.sweet :refer :all])
+  (:require [clojure.test :refer :all])
+  (:require [clojure.set :refer :all])
+  )
+
+;Day 1
+(defn is-binary-tree
+  "determines whether the argument sequence is a binary tree or not"
+  [x]
+  (if (not (coll? x))
+    (if (= x false) false true)
+    (if (= (count x) 3)
+      (and (is-binary-tree (first (rest x))) (is-binary-tree (last x)))
+      false))
+  )
+
+(deftest test-is-binary-tree
+  (is (= (is-binary-tree '(:a (:b nil nil) nil)) true))
+  (is (= (is-binary-tree '(:a (:b nil nil))) false))
+  (is (= (is-binary-tree [1 nil [2 [3 nil nil] [4 nil nil]]]) true))
+  (is (= (is-binary-tree [1 [2 nil nil] [3 nil nil] [4 nil nil]]) false))
+  (is (= (is-binary-tree [1 [2 [3 [4 nil nil] nil] nil] nil]) true))
+  (is (= (is-binary-tree [1 [2 [3 [4 false nil] nil] nil] nil]) false))
```

Comment:
I don't understand why it (line 24) should be false. And it made you write line 12, which otherwise is not needed.